### PR TITLE
Implement labels on Elastic IPs

### PR DESCRIPTION
**Pull request by PhilippeChepy**
Implement labels on Elastic IP resource and datasource.

```
➜ TF_ACC=true go test -timeout 600s -run ^TestAccDataSourceElasticIP$ github.com/exoscale/terraform-provider-exoscale/exoscale
ok      github.com/exoscale/terraform-provider-exoscale/exoscale        17.130s
```

```
➜ TF_ACC=true go test -timeout 600s -run ^TestAccResourceElasticIP$ github.com/exoscale/terraform-provider-exoscale/exoscale
ok      github.com/exoscale/terraform-provider-exoscale/exoscale        35.596s
```

### CODE REVIEW

**Comment by PhilippeChepy** [comment]
This branch needs https://github.com/exoscale/egoscale/pull/566 to be merged first

File: exoscale/resource_exoscale_elastic_ip.go [hunk 1]
```diff
@@ -213,6 +219,14 @@ func resourceElasticIPCreate(ctx context.Context, d *schema.ResourceData, meta i
 		}
 
 		elasticIP.Healthcheck = &elasticIPHealthcheck
+
+		if l, ok := d.GetOk(resElasticIPAttrLabels); ok {
+			labels := make(map[string]string)
+			for k, v := range l.(map[string]interface{}) {
```
Comment (kobajagi) [inline exoscale/resource_exoscale_elastic_ip.go:225]: Casting to `map[string]string` should work.

File: exoscale/resource_exoscale_elastic_ip.go [hunk 2]
```diff
@@ -279,6 +293,15 @@ func resourceElasticIPUpdate(ctx context.Context, d *schema.ResourceData, meta i
 
 	var updated bool
 
+	if d.HasChange(resElasticIPAttrLabels) {
+		labels := make(map[string]string)
+		for k, v := range d.Get(resElasticIPAttrLabels).(map[string]interface{}) {
+			labels[k] = v.(string)
+		}
+		elasticIP.Labels = &labels
```
Comment (kobajagi) [inline exoscale/resource_exoscale_elastic_ip.go:301]: I don't see the need to copy labels by value. Assuming my previous suggestion works (casting to `map[string]string`), we can just set pointer to that. I could be wrong though, to be tested.

Comment (kobajagi) [reply exoscale/resource_exoscale_elastic_ip.go:301]: Same snippet is all over our codebase, but if simplification works we might as well start code cleanup from this.

**Review by kobajagi — APPROVED** [review]
Great work!

File: exoscale/resource_exoscale_elastic_ip.go (hunk 1 above)
Comment (PhilippeChepy) [reply exoscale/resource_exoscale_elastic_ip.go:225]: It should (intuitively), but when trying to cast  `l` directly into a `map[string]string` it leads to:

```
panic: interface conversion: interface {} is map[string]interface {}, not map[string]string
```

Comment (kobajagi) [reply exoscale/resource_exoscale_elastic_ip.go:225]: I guess i was wrong. Thanks for checking.

**Review by jessicatoscani — APPROVED** [review]

**Pull request merged**